src/service/resources-impl.js

```diff
@@ -866,12 +869,23 @@ export class Resources {
         const resource = request.resource;
         const box = resource.getLayoutBox();
         const iniBox = resource.getInitialLayoutBox();
-        const diff = request.newHeight - box.height;
+        const margins = resource.getLayoutMargins();
```

Comment:
For simplicity, shouldn't we just do `diff = (newHeight + newMargin) - (oldHeight + oldMargin)` and operate with that as before? Why I'm asking is that naively, margin changes are visually very similar to padding changes and paddings would be captured inside the height. So it'd seem that the `diff` math should be the same?